lib/active_merchant/billing/gateways/mundipagg.rb

```diff
@@ -0,0 +1,287 @@
+        end
+      end
+
+      def add_payment(post, payment, options)
+        post[:customer][:name] = payment.name if post[:customer]
+        post[:customer_id] = parse_auth(payment)[0] if payment.is_a?(String)
+        post[:payment] = {}
+        post[:payment][:gateway_affiliation_id] = @options[:gateway_id]
+        post[:payment][:metadata] = { mundipagg_payment_method_code: '1' } if test?
+        if voucher?(payment)
+          add_voucher(post, payment, options)
+        else
+          add_credit_card(post, payment, options)
+        end
+      end
+
+      def add_credit_card(post, payment, options)
+        post[:payment][:payment_method] = 'credit_card'
+        post[:payment][:credit_card] = {}
+        if payment.is_a?(String)
+          post[:payment][:credit_card][:card_id] = parse_auth(payment)[1]
+        else
+          post[:payment][:credit_card][:card] = {}
+          post[:payment][:credit_card][:card][:number] = payment.number
+          post[:payment][:credit_card][:card][:holder_name] = payment.name
+          post[:payment][:credit_card][:card][:exp_month] = payment.month
+          post[:payment][:credit_card][:card][:exp_year] = payment.year
+          post[:payment][:credit_card][:card][:cvv] = payment.verification_value
+          post[:payment][:credit_card][:card][:billing_address] = build_billing_address(options)
+        end
+      end
+
+      def add_voucher(post, payment, options)
+        post[:currency] = 'BRL'
+        post[:payment][:payment_method] = 'voucher'
+        post[:payment][:voucher] = {}
+        post[:payment][:voucher][:card] = {}
+        post[:payment][:voucher][:card][:number] = payment.number
+        post[:payment][:voucher][:card][:holder_name] = payment.name
+        post[:payment][:voucher][:card][:holder_document] = options[:holder_document]
+        post[:payment][:voucher][:card][:exp_month] = payment.month
+        post[:payment][:voucher][:card][:exp_year] = payment.year
+        post[:payment][:voucher][:card][:cvv] = payment.verification_value
+        post[:payment][:voucher][:card][:billing_address] = build_billing_address(options)
+      end
+
+      def voucher?(payment)
+        return false if payment.is_a?(String)
+        %w[sodexo vr].include? card_brand(payment)
+      end
+
+      def headers
+        {
+          'Authorization' => 'Basic ' + Base64.strict_encode64("#{@options[:api_key]}:"),
+          'Content-Type' => 'application/json',
+          'Accept' => 'application/json'
+        }
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def url_for(action, auth = nil)
+        url = live_url
+        return "#{url}customers/#{auth}/cards" if action == 'store'
+        return "#{url}customers" if action == 'customer'
+        url = "#{url}charges/"
+        url =  "#{url}#{auth}/" if %w[refund void capture].include? action
+        return "#{url}capture/" if action == 'capture'
+        url
+      end
```

Comment:
Determining which URL to use would be better suite with a `case` statement here. Additionally, if the trailing slash is removed from https://github.com/activemerchant/active_merchant/pull/2791/files#diff-d6e4d483fbc1ae8bde58049c05c40018R4, it'll be easier to read which endpoint is being used:
```
"#{url}/customers/#{auth}/cards"
```
vs
```
"#{url}customers/#{auth}/cards"
```